src/index.ts

```diff
@@ -45,23 +47,51 @@ async function main() {
   });
   Prometheus.collectDefaultMetrics({ register });
 
-  const client = new Client({
-    url: conf.mercuryGraphQL,
-    exchanges: [fetchExchange],
-    fetchOptions: () => {
-      return {
-        headers: { authorization: `Bearer ${conf.mercuryKey}` },
-      };
-    },
-  });
-  // we need a second client because the authenticate muation does not ignore the current jwt
-  const renewClient = new Client({
-    url: conf.mercuryGraphQL,
-    exchanges: [fetchExchange],
-  });
+  const graphQlEndpoints = {
+    TESTNET: conf.mercuryGraphQLTestnet,
+    PUBLIC: conf.mercuryGraphQLPubnet,
+  };
+
+  const backends = {
+    TESTNET: conf.mercuryBackendTestnet,
+    PUBLIC: conf.mercuryBackendPubnet,
+  };
+
+  function renewClientMaker(network: NetworkNames) {
```

Comment:
super NIT style thing: is the reason for using `function` rather than `const renewClientMaker = () =>`  here to hoist? If not, would probably better to follow the coding conventions in the rest of the repo

Reply:
oh that's just oversight, no hoisting needed. Fixed in 1ad7aba92e0c854149bfad7bc6c9ded2332cae61